src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNetCoreMvc2Integration.cs

```diff
@@ -250,6 +256,66 @@ public AspNetCoreMvc2Integration(object actionDescriptor, object httpContext)
             }
         }
 
+        /// <summary>
+        /// Wrapper method used to catch unhandled exceptions in the incoming request pipeline for Microsoft.AspNetCore.Mvc.Core
+        /// </summary>
+        /// <param name="context">The DiagnosticSource that this extension method was called on.</param>
+        [InterceptMethod(
+            CallerAssembly = "Microsoft.AspNetCore.Mvc.Core",
+            TargetAssembly = "Microsoft.AspNetCore.Mvc.Core",
+            TargetType = ExceptionHookType)]
+        public static void Rethrow(object context)
+        {
+            AspNetCoreMvc2Integration integration = null;
+            const string methodName = nameof(Rethrow);
+
+            try
+            {
+                if (context.TryGetPropertyValue("HttpContext", out object httpContext))
+                {
+                    if (httpContext.TryGetPropertyValue("Items", out IDictionary<object, object> contextItems))
+                    {
+                        integration = contextItems?[HttpContextKey] as AspNetCoreMvc2Integration;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.ErrorExceptionForFilter($"Error accessing {nameof(AspNetCoreMvc2Integration)}.", ex);
+            }
+
+            Action<object> rethrow;
+
+            try
+            {
+                rethrow = RethrowCache.GetInterceptedMethod(
+                    assembly: Assembly.GetCallingAssembly(),
+                    owningType: ExceptionHookType,
+                    methodName: nameof(Rethrow),
+                    generics: Interception.EmptyTypes,
+                    parameters: Interception.TypeArray(context));
```

Comment:
This would be the syntax we would use to retrieve intercepted methods everywhere else. 
With this comes the type checks and caching, which should also prevent most of the signature conflict issues from ever being a problem.